### PR TITLE
Flutter 3.0 update, addresses api changes and fixes new linter complaints

### PR DESCRIPTION
**Pull request by frankpepermans**
Nothing special, mainly an update for Flutter 3.0:
- update pubspec to latest refs
- fix all linter complaints
- add docs to classes that were private before

### CODE REVIEW

File: lib/src/controllers/reader_mode_controller.dart [hunk 1]
```diff
@@ -159,18 +158,25 @@ class ReaderModeController extends ChangeNotifier {
   }
 }
 
-class _UriWithScrollPosition {
+/// Tuple class containing the [position] for a given [uri].
+class UriWithScrollPosition {
+  /// The url that was loaded
   final Uri uri;
```
Comment (MichaelKMalak) [inline lib/src/controllers/reader_mode_controller.dart:164]: Just to be sure.. in this case, if we still want `UriWithScrollPosition` to be private, we can simply put it in a different file and not export that file, right?

Comment (MichaelKMalak) [reply lib/src/controllers/reader_mode_controller.dart:164]: same with `AncestorWithLevel`, `RowsAndColumns`, `ProcessedHtml` and`GrabArticleResult`

**Review by MichaelKMalak — APPROVED** [review]
looks good 🚀  added a comment but no blocker.

File: lib/src/controllers/reader_mode_controller.dart (hunk 1 above)
Comment (frankpepermans) [reply lib/src/controllers/reader_mode_controller.dart:164]: yes, but it's open source, if people want to override certain API elements, they can actually do that now, so maybe they are best kept public throughout.

**Pull request merged**